src/sokoban.cpp

```diff
@@ -37,4 +43,54 @@ void Sokoban::locate_player() {
         }
     }
     throw std::invalid_argument("Player not found...");
+}
+
+bool Sokoban::make_move(int dy, int dx) {
+    char current_cell;
+    char next_cell;
+    //If the player is currently on a goal cell ,'+', then moves
+    //off of the goal cell, the current_cell will revert to an open cell
+    //or a goal cell.
+    if (_board[py][px] == '+') {
+        current_cell = '.';
+    }
+    else {
+        current_cell = ' ';
+    }
+
+    //If the player moves to a goal cell, then the next_cell
+    //will change the player's state to '+',
+    //if not moved to a goal cell then, '@'.
+    if (_board[py + dy][px + dx] == '.') {
+        next_cell = '+';
+    }
+    else {
+        next_cell = '@';
+    }
+
+    if(_board[py+dy][px+dx] == '.' || _board[py+dy][px+dx] == ' ') {
+        _board[py][px] = current_cell;
+        _board[py+dy][px+dx] = next_cell;
+        py += dy;
+        px += dx;
+        return true;
+    }
+    return false;
+}
+
+bool Sokoban::move(char direction) {
+    direction = toupper(direction);
+    if (direction == 'U') {
```

Comment:
This is fine for now, but if you use a map lookup, you can remove the `if`s entirely.

See also my suggestion about changing `char direction` to `Sokoban::Direction direction` on Discord. We can do this in the next PR though.